### PR TITLE
feat: support for generated presigner

### PR DESCRIPTION
**Pull request by wooj2**
Corresponding:
https://github.com/awslabs/aws-sdk-swift/pull/389

## Scope

- [x] New feature (non-breaking change which adds functionality)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: smithy-swift-codegen/src/main/kotlin/software/amazon/smithy/swift/codegen/integration/HttpProtocolClientGenerator.kt [hunk 1]
```diff
@@ -74,7 +59,14 @@ open class HttpProtocolClientGenerator(
             operations.forEach {
                 ServiceGenerator.renderOperationDefinition(model, symbolProvider, writer, operationsIndex, it)
                 writer.openBlock("{", "}") {
-                    renderMiddlewareExecutionBlock(operationsIndex, it)
+                    val operationStackName = "operation"
+                    val generator = MiddlewareExecutionGenerator(ctx, writer, httpBindingResolver, defaultContentType, httpProtocolCustomizable, operationStackName)
```
Comment (kneekey23) [inline smithy-swift-codegen/src/main/kotlin/software/amazon/smithy/swift/codegen/integration/HttpProtocolClientGenerator.kt:63]: so nice and clean!

File: smithy-swift-codegen/src/main/kotlin/software/amazon/smithy/swift/codegen/integration/MiddlewareExecutionGenerator.kt [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+        renderUriPath(httpTrait, pathBindings, writer, onError)
+
+        val operationErrorName = "${op.capitalizedName()}OutputError"
+        val inputShapeName = ServiceGenerator.getOperationInputShapeName(symbolProvider, opIndex, op)
+        val outputShapeName = ServiceGenerator.getOperationOutputShapeName(symbolProvider, opIndex, op)
+        writer.write("let context = \$N()", ClientRuntimeTypes.Http.HttpContextBuilder)
+        writer.swiftFunctionParameterIndent {
+            renderContextAttributes(op)
+        }
+        writer.write("var $operationStackName = OperationStack<$inputShapeName, $outputShapeName, $operationErrorName>(id: \"${op.camelCaseName()}\")")
```
Comment (kneekey23) [inline smithy-swift-codegen/src/main/kotlin/software/amazon/smithy/swift/codegen/integration/MiddlewareExecutionGenerator.kt:51]: this looks like a spot I forgot to make this a runtime type -> OperationStack

File: smithy-swift-codegen/src/main/kotlin/software/amazon/smithy/swift/codegen/integration/ProtocolGenerator.kt [hunk 3]
```diff
@@ -119,6 +119,10 @@ interface ProtocolGenerator {
      */
     fun generateProtocolClient(ctx: GenerationContext)
 
+    fun getProtocolHttpBindingResolver(ctx: ProtocolGenerator.GenerationContext): HttpBindingResolver =
+        HttpTraitResolver(ctx)
+
+    val httpProtocolCustomizable: HttpProtocolCustomizable
```
Comment (kneekey23) [inline smithy-swift-codegen/src/main/kotlin/software/amazon/smithy/swift/codegen/integration/ProtocolGenerator.kt:125]: love this change, as discussed!

**Review by kneekey23 — APPROVED** [review]
LGTM, didn't see really anything even nit picky. really nice work!

File: smithy-swift-codegen/src/main/kotlin/software/amazon/smithy/swift/codegen/integration/MiddlewareExecutionGenerator.kt (hunk 2 above)
Comment (wooj2) [reply smithy-swift-codegen/src/main/kotlin/software/amazon/smithy/swift/codegen/integration/MiddlewareExecutionGenerator.kt:51]: ah! on it.

**Pull request merged**